### PR TITLE
Added rendering of tag tourism=wilderness_hut

### PR DESCRIPTION
**Pull request by Ircama**

### CODE REVIEW

**Comment by Ircama** [comment]
Added rendering of tag tourism=wilderness_hut
Added specific symbols for basic_hut and wilderness_hut vs. generic amenity=shelter.

Notice that this drop shall be consistent with the other push related to zooms, where alpine_hut symbol (the main one) is shown at zoom>=13.

To summarize, the following different kind of huts have specific symbols now:

amenity=shelter, shelter_type=basic_hut (zoom>=13)
amenity=shelter (only shown at zoom >=16)
tourism=alpine_hut (zoom>=13)
tourism=wilderness_hut (not modified within this drop)

hut

File: project.yaml [hunk 1]
```diff
@@ -1666,11 +1666,12 @@ Layer:
             denomination,
             "generator:source",
             power_source,
+            tags->'shelter_type'::text AS shelter_type,
```
Comment (pnorman) [inline project.yaml]: We do not have hstore available so this does not work.

File: amenity-points.mss [hunk 2]
```diff
@@ -27,8 +27,20 @@
...
   }
 
+  [feature = 'tourism_wilderness_hut'][zoom >= 13],
+  [feature = 'amenity_shelter']['shelter_type' = 'basic_hut'][zoom >= 14],
   [feature = 'amenity_shelter'][zoom >= 16] {
     marker-file: url('symbols/shelter-14.svg');
+    [feature = 'tourism_wilderness_hut'] {
+    	marker-file: url('symbols/wilderness_hut.svg');
+    	marker-transform:"scale(0.6)";
+    	[zoom < 19] { marker-transform:"scale(0.5)"; }
```
Comment (pnorman) [inline amenity-points.mss]: Have the first lines be the conditions at low zoom, and then override for `zoom >= 19` if needed

File: symbols/basic_hut.svg [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <inkscape:grid type="xygrid" id="grid2383" visible="true" enabled="true" empspacing="2" snapvisiblegridlinesonly="true"/>
+  </sodipodi:namedview>
+  <defs id="defs4">
+    <inkscape:perspective sodipodi:type="inkscape:persp3d" inkscape:vp_x="0 : 526.18109 : 1" inkscape:vp_y="0 : 1000 : 0" inkscape:vp_z="744.09448 : 526.18109 : 1" inkscape:persp3d-origin="372.04724 : 350.78739 : 1" id="perspective10"/>
+    <inkscape:perspective id="perspective2461" inkscape:persp3d-origin="10 : 6.6666667 : 1" inkscape:vp_z="20 : 10 : 1" inkscape:vp_y="0 : 1000 : 0" inkscape:vp_x="0 : 10 : 1" sodipodi:type="inkscape:persp3d"/>
+    <inkscape:perspective id="perspective2466" inkscape:persp3d-origin="400 : 200 : 1" inkscape:vp_z="800 : 300 : 1" inkscape:vp_y="0 : 1000 : 0" inkscape:vp_x="0 : 300 : 1" sodipodi:type="inkscape:persp3d"/>
+  </defs>
+  <path style="fill:none;stroke:#0092da;stroke-width:4;stroke-linecap:butt;stroke-linejoin:round;stroke-miterlimit:4;stroke-opacity:1;stroke-dasharray:none" d="M 16,2 2,16 6,15.999304 6,30 l 20,0 0,-14 4,0 z" id="rect3055" inkscape:connector-curvature="0" sodipodi:nodetypes="cccccccc"/>
+</svg>
```
Comment (pnorman) [inline symbols/basic_hut.svg]: The SVGs need cleaning, I believe you can export a clean one.

File: amenity-points.mss [hunk 4]
```diff
@@ -27,8 +27,20 @@
     point-placement: interior;
   }
 
+  [feature = 'tourism_wilderness_hut'][zoom >= 13],
+  [feature = 'amenity_shelter']['shelter_type' = 'basic_hut'][zoom >= 14],
   [feature = 'amenity_shelter'][zoom >= 16] {
     marker-file: url('symbols/shelter-14.svg');
+    [feature = 'tourism_wilderness_hut'] {
+    	marker-file: url('symbols/wilderness_hut.svg');
+    	marker-transform:"scale(0.6)";
```
Comment (pnorman) [inline amenity-points.mss]: Can marker-width/marker-height do this, or does it need to be a transform?

Comment (pnorman) [inline amenity-points.mss]: The marker should be sized an appropriate size to not need a transform. Also, the same SVG will not look good at two different scales.

**Comment by pnorman** [comment]
2.1k usages is towards the low end, but on the other hand this is a feature which will normally be encountered in remote areas.

There are a lot of commits in this PR, could you rebase them into one?

**Comment by Ircama** [comment]
> 2.1k usages is towards the low end, but on the other hand this is a feature which will normally be encountered in remote areas.

Could you please better explain?

> We do not have hstore available so this does not work.

I expected it was available, it looks already present in openstreetmap-carto.style, but probably I misunderstood a comment by @HolgerJeromin.

**Comment by HolgerJeromin** [comment]
> There are a lot of commits in this PR, could you rebase them into one?

If i am not wrong squashing commits is now available in the merge interface.

**Comment by matkoniecz** [comment]
> If i am not wrong squashing commits is now available in the merge interface. 

It is not enabled by default. See #2103

**Comment by matkoniecz** [comment]
> Could you please better explain?

2.1k is above "low usage, therefore rendering support is certainly a bad idea", but still it is not popular tag. So it is slight nudge toward not rendering it.

"on the other hand this is a feature which will normally be encountered in remote areas." - so it is unlikely to compete with already rendered icons, so it slight nudge toward rendering it.

**Comment by imagico** [comment]
Since alpine_hut and shelter use a common icon design it is not a good idea to have a different design for this which is logically between the other two.

Apart from that the idea itself seems fine.

**Comment by Ircama** [comment]
> There are a lot of commits in this PR, could you rebase them into one?

Please, suggest the correct git commands, thanks. I usually perform this:
```
git pull
<edit>
git commit -a
git push
```

Meanwhile I revised the code that now should reflect feedbacks by @pnorman:

- Usage of marker-width/marker-height
- SVGs are cleaned now

Icon usage:

- alpine_hut: symbols/alpinehut.p.16.png
- wilderness_hut: symbols/wilderness_hut.svg
- generic shelter: symbols/shelter-14.svg
- amenity=shelter with shelter_type=basic_hut: symbols/basic_hut.svg

@matkoniecz thanks for explanation.

Just in Italy we now have 774 huts which are managed by CAI (Italian alpine club; there are many more active huts in Italy so). Let's suppose that there might be at least two wilderness huts per alpine hut. We could imagine that the current figure of 1452 nodes + 688 ways cannot reflect the entire planet (notice also that they should be represented as building, not just nodes). Once again, unfortunately geographical information related to scarcely populated areas in OSM has currently no quality at all. Could its cause be the same mentioned in #2138? (That is, these kind of tags are not rendered). In my opinion with OSM people are currently more keen to add urban and suburban data ("OSM=open street") and data that can be rendered ("OSM=open map"). To try to fill this gap (let's say, OSM to become open planet data), IMHO it should be appropriate to enable many more elements on the OSM map. Also, as @imagico was mentioning in #2138 (and @drgweber in #1148), we would need an effective communication program, including well defined style guides to drive mappers on “remote areas”, which are currently green field for them and where business digital maps are slow to enter (market failure zones).

**Comment by matkoniecz** [comment]
> Please, suggest the correct git commands, thanks. I usually perform this:

See for example http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Note that "A word of caution: Only do this on commits that haven’t been pushed an external repository." does not apply here (as it may safely assumed that nobody pulled from this branch and based their work on this changes).

**Comment by Ircama** [comment]
This PR will be closed as superseded by #2145

**Pull request closed without merging**